docs/chains/deps-and-resources.mdx

```diff
@@ -0,0 +1,192 @@
+        # This is one of the main things that we've added. This shows off using
+        # the Model Cache feature of Truss to cache the weights of the model at
+        # build time.
+        assets=chains.Assets(cached=[
+            truss_config.ModelRepo(
+                repo_id=MISTRAL_HF_MODEL, allow_patterns=["*.json", "*.safetensors", ".model"]
+            )
+        ]),
+    )
+    default_user_config = MistraLLMConfig(hf_model_name=MISTRAL_HF_MODEL)
+
+    def __init__(
+        self,
+        context: chains.DeploymentContext[MistraLLMConfig] = chains.provide_context(),
+    ) -> None:
+        super().__init__(context)
+        # Note that the code in the __init__ and run methods are
+        import torch
+        import transformers
+
+        model_name = self.user_config.hf_model_name
+
+        self._model = transformers.AutoModelForCausalLM.from_pretrained(
+            model_name,
+            torch_dtype=torch.float16,
+            device_map="auto",
+        )
+        self._tokenizer = transformers.AutoTokenizer.from_pretrained(
+            model_name,
+            device_map="auto",
+            torch_dtype=torch.float16,
+        )
+
+        self._generate_args = {
+            "max_new_tokens": 512,
+            "temperature": 1.0,
+            "top_p": 0.95,
+            "top_k": 50,
+            "repetition_penalty": 1.0,
+            "no_repeat_ngram_size": 0,
+            "use_cache": True,
+            "do_sample": True,
+            "eos_token_id": self._tokenizer.eos_token_id,
+            "pad_token_id": self._tokenizer.pad_token_id,
+        }
+
+    def run(self, prompt: str) -> str:
+        import torch
+
+        formatted_prompt = f"[INST] {data} [/INST]"
+        input_ids = self._tokenizer(
+            formatted_prompt, return_tensors="pt"
+        ).input_ids.cuda()
+        with torch.no_grad():
+            output = self._model.generate(inputs=input_ids, **self._generate_args)
+            result = self._tokenizer.decode(output[0])
+        return result
+```
+
+# Chaining it all together
+
+Once we have our LLM that can produce results, we can chain it with our Chainlet that generates poems.
+
+```python
+import truss_chains as chains
+
+class PoemGenerator(chains.ChainletBase):
+    remote_config = chains.RemoteConfig(
+        docker_image=chains.DockerImage(
+            pip_requirements=["git+https://github.com/basetenlabs/truss.git@9be68f563d735289128cd3c8853bdf62ec03cef3"]
+        )
+    )
+
+    def __init__(
+        self,
+        context: chains.DeploymentContext = chains.provide_context(),
+        mistral_llm: MistralLLM = chains.provide(MistralLLM),
+    ) -> None:
+        super().__init__(context)
+        self._mistral_llm = mistral_llm
+
+    def run(self, words: list[str]) -> list[str]:
+        results = []
+        for word in words:
+            poem = self._mistral_llm.run(f"Generate a poem about: {word}")
+            results.append(poem)
+        return results
+```
+
```

Comment:
Optional: either add here (or leave a ref to) the option to debug/dev locally with a mock of mistral that implements the same protocol.